proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/ProxyNodeLogger.java

```diff
@@ -16,7 +16,36 @@ public void addContext(ProxyNodeMDCKey key, String value) {
         MDC.put(key.name(), value);
     }
 
-    public void addContext(Exception exception) {
+    public void info(String message) {
+        log(Level.INFO, message);
+    }
+
+    public void warning(String message) {
+        log(Level.WARNING, message);
+    }
+
+    public void error(String message) {
+        log(Level.SEVERE, message);
+    }
+
+    public void logException(Exception exception, Level level, String message) {
+        addExceptionContext(exception);
+        log(level, message);
+    }
+
+    private void log(Level level, String message) {
+        logWithCallingFrame(level, () -> message, null);
+    }
+
+    private void log(Level level, String message, Exception exception) {
+        logWithCallingFrame(level, () -> message, exception);
+    }
+
+    private void log(Level level, Supplier<String> message) {
+        logWithCallingFrame(level, message, null);
+    }
+
+    private void addExceptionContext(Exception exception) {
```

Comment:
Where does the context added in the method get removed? Shouldn't we be cleaning up everything from the MDC after we log?

Reply:
In the `ProxyNodeLogging` filter:
```
for (ProxyNodeMDCKey key : ProxyNodeMDCKey.values()) {
            MDC.remove(key.name());
        }
```

We clean everything up before sending the response back

Reply:
Oh nice.